player/lua/console.lua

```diff
@@ -232,6 +239,23 @@ function ass_escape(str)
     return mp.command_native({'escape-ass', str})
 end
 
+local function calculate_max_log_lines()
+    if not mp.get_property_native('vo-configured') then
+        -- Subtract 1 for the input line and for each line in the status line.
+        -- This does not detect wrapped lines.
+        return mp.get_property_native('term-size/h', 24) - 2 -
+               select(2, mp.get_property('term-status-msg'):gsub('\\n', ''))
+    end
+
+    -- Subtract 1.5 to account for the input line.
+    return math.floor(mp.get_property_native('osd-height')
+                      / mp.get_property_native('display-hidpi-scale', 1)
+                      / opts.scale
+                      * (1 - global_margins.t - global_margins.b)
+                      / opts.font_size
+                      - 1.5)
```

Comment:
Shouldn't this 1.5 be scaled by font size or something?

Reply:
We want to subtract 1 line for the input line whatever is its font size and we also add 0.5 to reduce chances of the log being clipped, because as far as I know there is no sure way to get libass' text height other than `compute_bounds` which is slow, e.g. Japanese text is taller and can easily be clipped (the first line of `show-text ${playlist}` is also often clipped for me). I changed `math.ceil` to `math.floor` to mitigate this as clipping the default log was fine but it is annoying when you're using it as a selector.

Reply:
ok